src/exec/input.rs

```diff
@@ -5,13 +5,13 @@ use crate::filesystem::strip_current_dir;
 
 /// Removes the parent component of the path
 pub fn basename(path: &Path) -> &OsStr {
-    path.file_name().unwrap_or_else(|| path.as_os_str())
+    path.file_name().unwrap_or(path.as_os_str())
```

Comment:
It looks like this introduces a clippy warning? Or are you running a different clippy version offline?

Reply:
My local clippy version is 0.1.59. And the code quality github action passed. What is the warning it introduces?

Reply:
Strange. I'm seeing this inline annotation on Githubs diff view:

![image](https://user-images.githubusercontent.com/4209276/156717446-629c8337-693f-487b-81b1-ae47c6e65ee1.png)

Reply:
ah! the warning you see is from the minimum supported rust version check.

I found the commit that changed this in clippy: https://github.com/rust-lang/rust-clippy/pull/7639/files#diff-bada3dbc6599bd1411fea3afe759dbb229221e3d62d46bc8c835d5cb731e81b8

So, it now recommends using `unwrap_or` because it thinks `as_os_str()` is cheap (and it is) as of rust 1.59, but on older versions, it warned that it should use `unwrap_or_else`.